plasmapy/diagnostics/langmuir.py

```diff
@@ -1,153 +1,1073 @@
+
+    def check_validity(self):
+        r"""Check the unit and value validity of the characteristic."""
+
+        utils._check_quantity(self.bias, 'bias', str(self), u.V,
+                              can_be_negative=True, can_be_complex=False,
+                              can_be_inf=False, can_be_nan=True)
+
+        utils._check_quantity(self.current, 'bias', str(self), u.A,
+                              can_be_negative=True, can_be_complex=False,
+                              can_be_inf=False, can_be_nan=True)
+
+    def get_padded_limit(self, padding, log=False):
+        r"""Return the limits of the current range for plotting, taking into
+        account padding. Matplotlib lacks this functionality.
+
+        Parameters
+        ----------
+        padding : float
+            The padding ratio as a float between 0.0 and 1.0.
+
+        log : boolean, optional
+            If True the calculation will be performed on a logarithmic scale.
+            Default is False.
+
+        """
+
+        ymax = np.max(self.current).to(u.A).value
+
+        if(log):
+            ymin = np.min(np.abs(
+                    self.current[self.current != 0])).to(u.A).value
+            return [ymin * 10**(-padding * np.log10(ymax / ymin)),
+                    ymax * 10**(padding * np.log10(ymax / ymin))] * u.A
+        else:
+            ymin = np.min(self.current).to(u.A).value
+            return [ymin - padding * (ymax - ymin),
+                    ymax + padding * (ymax - ymin)] * u.A
+
+    def plot(self):
+        r"""Plot the characteristic in matplotlib."""
+
+        with quantity_support():
+            plt.figure()
+            plt.scatter(self.bias, self.current,
+                        marker='.', color='k')
+            plt.title("Probe characteristic")
+
+
+@utils.check_quantity({'probe_area': {'units': u.m**2,
+                                      'can_be_negative': False,
+                                      'can_be_complex': False,
+                                      'can_be_inf': False,
+                                      'can_be_nan': False}})
+def swept_probe_analysis(probe_characteristic, probe_area, gas,
+                         bimaxwellian=False, visualize=False,
+                         plot_electron_fit=False, plot_EEDF=False):
+    r"""Attempt to perform a basic swept probe analysis based on the provided
+    characteristic and probe data. Suitable for single cylindrical probes in
+    low-pressure DC plasmas, since OML is applied.
 
     Parameters
     ----------
-    potential_sweep : ~astropy.units.Quantity
-        The swept electric potential applied to the probe in units
-        convertible to V.
+    probe_characteristic : Characteristic
+        The swept probe characteristic that is to be analyzed.
+
+    probe_area : Quantity
+        The area of the probe exposed to plasma in units convertible to m^2.
+
+    gas : float
+        The (mean) mass of the background gas in atomic mass units (amu).
```

Comment:
I think `gas` here could potentially be a `Quantity`! We could set the  unit for that in the decorator to amu, and it'd turn floats into amu by default.